### PR TITLE
Update shared components in `BasicLTILaunchApp`; convert to TSX

### PR DESCRIPTION
**Pull request by lyzadanger**
Part of #5013

The only component updated was `FullScreenSpinner` => `SpinnerOverlay`, which is an exact replacement (no visual changes).

### CODE REVIEW

File: lms/static/scripts/frontend_apps/components/BasicLTILaunchApp.tsx [hunk 1]
```diff
@@ -102,21 +105,25 @@ export default function BasicLTILaunchApp() {
...
    *
-   * @param {Error} error - Error object from request.
-   * @param {ErrorState} state
-   * @param {boolean} [retry=true] - Can the request be retried?
-   * @param {string} [authURL] -
-   *   Authorization URL association with the API request
+   * @param error - Error object from request
+   * @param state
+   * @param [retry=true] - Can the request be retried?
+   * @param [authURL] - Authorization URL association with the API request
```
Comment (lyzadanger) [inline lms/static/scripts/frontend_apps/components/BasicLTILaunchApp.tsx:111]: I'm not sure what the best JSDoc approach is when one just wants to add a note about the _fourth_ parameter :).

Comment (robertknight) [reply lms/static/scripts/frontend_apps/components/BasicLTILaunchApp.tsx:111]: You can add `@param`s for just the arguments where you need to provide additional information beyond the name and type.

File: lms/static/scripts/frontend_apps/components/BasicLTILaunchApp.tsx [hunk 2]
```diff
@@ -102,21 +105,25 @@ export default function BasicLTILaunchApp() {
...
+  ) => {
     // Here we always set the authorization URL, but we could improve UX by
     // not setting it if the problem is not related to authorization (eg.
     // a network fetch error).
     setAuthURL(authURL || null);
 
     if (isLTILaunchServerError(error)) {
       setError(error);
-      setErrorState(/** @type {ErrorState} */ (error.errorCode));
+      setErrorState(error.errorCode as ErrorState);
```
Comment (robertknight) [inline lms/static/scripts/frontend_apps/components/BasicLTILaunchApp.tsx:126]: Is this just a non-null cast, if so you can use `!`.

**Review by robertknight — APPROVED** [review]

File: lms/static/scripts/frontend_apps/components/BasicLTILaunchApp.tsx (hunk 1 above)
Comment (lyzadanger) [reply lms/static/scripts/frontend_apps/components/BasicLTILaunchApp.tsx:111]: To confirm: It's OK to omit params that don't need comments? So, like:

```
@param [authURL] - Authorization URL association with the API request
```

would be OK on its own?

File: lms/static/scripts/frontend_apps/components/BasicLTILaunchApp.tsx (hunk 2 above)
Comment (lyzadanger) [reply lms/static/scripts/frontend_apps/components/BasicLTILaunchApp.tsx:126]: It's not :)

**Pull request merged**